libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewApi.kt

```diff
@@ -73,36 +61,115 @@ class NavigationViewApi internal constructor(
      */
-    fun isReplayEnabled(): Boolean {
-        return store.state.value.tripSession.isReplayEnabled
-    }
+    abstract fun isReplayEnabled(): Boolean
+
+    /**
+     * Clear Route data and request [NavigationView] to enter Free Drive state.
+     *
+     * [NavigationViewListener.onFreeDrive] or [NavigationViewListener.onDestinationPreview]
+     * (depending if destination has been set) will be called once [NavigationView] enters
+     * Free Drive state.
+     */
+    abstract fun startFreeDrive()
+
+    /**
+     * Request [NavigationView] to enter Route Preview state.
+     *
+     * [NavigationViewListener.onRoutePreview] will be called once [NavigationView] enters
+     * Route Preview state.
+     *
+     * @throws IllegalStateException if either Destination or Preview Routes are not set.
+     */
+    @Throws(IllegalStateException::class)
+    abstract fun startRoutePreview()
+
+    /**
+     * Request [NavigationView] to enter Active Navigation state.
+     *
+     * [NavigationViewListener.onActiveNavigation] will be called once [NavigationView] enters
+     * Active Navigation state.
+     *
+     * @throws IllegalStateException if either Destination or Preview Routes are not set.
+     */
+    @Throws(IllegalStateException::class)
+    abstract fun startNavigation()
+
+    /**
+     * Request [NavigationView] to enter Arrival state.
+     *
+     * [NavigationViewListener.onArrival] will be called once [NavigationView] enters
+     * Arrival state.
+     */
+    abstract fun startArrival()
+
+    /**
+     * Clear Destination and Route data, and update [NavigationView] to Free Drive state.
+     */
+    abstract fun endNavigation()
+}
+
+/**
+ * Sets a [destination], preview [routes] and request [NavigationView] to enter Route Preview state.
+ *
+ * It is a shorthand for calling the following with non-empty [routes].
+ * ```
+ *     setDestination(point)
+ *     setPreviewRoutes(routes)
+ *     startRoutePreview()
+ * ```
+ *
+ * @throws IllegalArgumentException if the [routes] argument is an empty list.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+@Throws(IllegalArgumentException::class)
+fun NavigationViewApi.startRoutePreview(destination: Point, routes: List<NavigationRoute>) {
+    require(routes.isNotEmpty()) { "routes cannot be empty" }
+
+    setDestination(destination)
+    setPreviewRoutes(routes)
+    startRoutePreview()
+}
+
+/**
+ * Sets a [destination], preview [routes] and request [NavigationView] to enter Active Navigation
+ * state.
+ *
+ * It is a shorthand for calling the following with non-empty [routes].
+ * ```
+ *     setDestination(point)
+ *     setPreviewRoutes(routes)
+ *     startNavigation()
+ * ```
+ *
+ * @throws IllegalArgumentException if the [routes] argument is an empty list.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+@Throws(IllegalArgumentException::class)
+fun NavigationViewApi.startNavigation(destination: Point, routes: List<NavigationRoute>) {
```

Comment:
Do we need destination as an input? We can deduce it internally and set it to the last point from `routes`.